src/k8s/api/v1/types.go

```diff
@@ -58,6 +71,32 @@ func BootstrapConfigFromMap(m map[string]string) (*BootstrapConfig, error) {
 	return config, nil
 }
 
+type JoinClusterConfig struct {
```

Comment:
I would recommend going further and splitting this to two separate `ControlPlaneNodeJoinConfig` and `WorkerNodeJoinConfig` structs. Silently ignoring a `APIServerCert` (or even misleading consumers to provide it) for worker nodes doesn't seem appropriate